### DISCUSSION TITLE
Tooltips in Labels don't trigger

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#1123] [Q&A] [answered]
Hello team, maybe I'm doing something wrong but nesting a Tooltip inside of a Label doesn't seem to let it trigger (focuses on the label instead). 

```tsx
<Label htmlFor="addressRequirement">
  <Box display="flex" alignItems="center">
    <Text as="span" marginRight="space10">Address requirements</Text>
    <Tooltip text="Some local authorities require you to provide an address before purchasing a phone number.">
      <InformationIcon decorative={false} title="Open Tooltip" display="block" />
    </Tooltip>
  </Box>
</Label>
```

can't see my mouse, but hovering over the icon doesn't open up the tooltip:

![image](https://user-images.githubusercontent.com/6936736/106184239-1e65c580-6156-11eb-8d08-3d0c1ab923fd.png)

**Comment by TheSisb** [comment]
Hi @vnguyen94,

This seems to be a bug because we don't currently forwardRefs on icons, and so you're probably seeing this error in console:
![image](https://user-images.githubusercontent.com/1592327/106192451-33545000-617a-11eb-9d62-a1c03bf4c145.png)
https://codesandbox.io/s/hidden-bird-2s6n6?file=/src/index.tsx

I'll add a ticket to fix this in our next sprint, as all of our components should be forwarding refs.

**Reply by SiTaggart** [reply]
Buuuuuuuuuuuuuuuuut as per the tooltip guidelines, you should be wrapping your icon with a focusable element https://paste.twilio.design/components/tooltip#focusable-element

**Reply by vankhoawin** [reply]
Is there something in the roadmap about making this a Pattern? I'm sure this is a very common thing throughout all teams (what should be the focusable element? button, link, something else?)

**Reply by vankhoawin** [reply]
Because I don't want to add any additional styling (or functionality like linking), I just want an icon with a tooltip

**Reply by vankhoawin** [reply]
> so you're probably seeing this error in console:

I actually don't see that error 😬 

edit: i do see that error in your csb link, but not in my application

**Reply by TheSisb** [reply]
Good point Simon. @vnguyen94 our guidance is as follows https://paste.twilio.design/components/tooltip#accessibility.  It may have been done in the past, but that pattern is bad for a11y so we should work towards changing it.

> edit: i do see that error in your csb link, but not in my application

😕 Not sure why that is tbh.

**Reply by vankhoawin** [reply]
actually looking back at code that you all helped me with before, we're doing this:
```tsx
        <Tooltip text="some text"}>
          <Button size="reset" variant="reset">
            <Icon decorative={false} title="some title" />
          </Button>
        </Tooltip>
```
if this is still the recommended way, i'll go ahead and keep using this pattern

**Reply by vankhoawin** [reply]
.... and that was the issue! thank you, closing :)

![image](https://user-images.githubusercontent.com/6936736/106195640-44df2d00-6165-11eb-9564-936a73e0d5c9.png)

**Comment by TheSisb** [comment, marked as answer]
Link to a more full answer here: https://github.com/twilio-labs/paste/discussions/683#discussioncomment-56645
